var/spack/repos/builtin/packages/patchelf/package.py

```diff
@@ -18,3 +24,29 @@ class Patchelf(AutotoolsPackage):
     version('0.10', sha256='b2deabce05c34ce98558c0efb965f209de592197b2c88e930298d740ead09019')
     version('0.9',  sha256='f2aa40a6148cb3b0ca807a1bf836b081793e55ec9e5540a5356d800132be7e0a')
     version('0.8',  sha256='14af06a2da688d577d64ff8dac065bb8903bbffbe01d30c62df7af9bf4ce72fe')
+
+    def _setup_hello(self):
+        src = os.path.join(os.path.dirname(__file__), 'test', 'hello')
+        testdir = tempfile.mkdtemp()
+        dest = os.path.join(testdir, 'hello')
+        shutil.copy(src, dest)
+        return dest
+
+    def test(self):
```

Comment:
I don't think this will be run, you need to add:
```python
@run_after('install')
@on_package_attributes(run_tests=True)
def test(self):
```

Reply:
> I don't think this will be run, you need to add:
> 
> ```python
> @run_after('install')
> @on_package_attributes(run_tests=True)
> def test(self):
> ```

Thanks Adam.

This work is actually part of a collaboration with @becker33 -- hence merging into `features/spack-test`.  It has been tested using that branch's `spack test <package>`.  I'll leave it to Greg to explain his plan and goals.

Reply:
The short version is, the current `spack test` command would move to `spack unit-test`, and we would introduce a `spack test` command that can run either smoke tests (like this one) or capability tests using a more advanced testing framework. This would be tests that could be run against any existing install of the package, and would not need any particular coordination with the install process.

Reply:
@becker33 How will that work for packages that already have a `test` method for build-time tests? This is the case for the entire `PythonPackage` build system, for example. Currently we tend to use test/check for build-time tests and install_test/installcheck for install-time tests.

Reply:
We'll have to rename those to `build_test` or `check` I suppose.

The build/install time tests are nice, but we can't use them for the same sort of system testing that we need because of how tied into each individual package's build system they are. The goal here is to do something more general (although possibly less complete).